src/measure_theory/interval_integral.lean

```diff
@@ -1336,22 +1336,44 @@ begin
 end
 
 /-- Fundamental theorem of calculus-2: If `f : ℝ → E` is continuous on `[a, b]` and has a derivative
-  at `f' x` for all `x` in `[a, b)`, and `f'` is continuous on `[a, b]` and measurable, then
-  `∫ y in a..b, f' y` equals `f b - f a`. -/
+  at `f' x` for all `x` in `(a, b)`, and `f'` is continuous on `[a, b]`, then `∫ y in a..b, f' y`
+  equals `f b - f a`. -/
 theorem integral_eq_sub_of_has_deriv_at' (hcont : continuous_on f (interval a b))
-  (hderiv : ∀ x ∈ Ico (min a b) (max a b), has_deriv_at f (f' x) x)
+  (hderiv : ∀ x ∈ Ioo (min a b) (max a b), has_deriv_at f (f' x) x)
   (hcont' : continuous_on f' (interval a b)) :
   ∫ y in a..b, f' y = f b - f a :=
-integral_eq_sub_of_has_deriv_right hcont (λ x hx, (hderiv x hx).has_deriv_within_at) hcont'
+begin
+  refine integral_eq_sub_of_has_deriv_right hcont _ hcont',
+  intros y hy,
+  rw [Ico, mem_set_of_eq, le_iff_lt_or_eq, or_and_distrib_right, ← mem_Ioo] at hy,
+  cases hy,
+  { exact (hderiv y hy).has_deriv_within_at },
+  { refine has_deriv_at_interval_left_endpoint_of_tendsto_deriv
+      (λ x hx, (hderiv x hx).has_deriv_within_at.differentiable_within_at)
+        ((hcont y (Ico_subset_Icc_self (mem_Ico.mpr ⟨hy.1.le, hy.2⟩))).mono Ioo_subset_Icc_self)
+          _ _,
```

Comment:
Given that this `refine` stretches over multiple lines, I would just write it as 
```
refine has_deriv_at_interval_left_endpoint_of_tendsto_deriv _ _ _ _,
````
and let the two long terms be `{ exact .... },` on their own lines.

Reply:
If I leave all four arguments blank (i.e. if I don't include at least one of these terms in the `refine`) then it generates an extra goal, so I don't know this is worthwhile. I was also able to shorten the second of these terms considerably, so I think this way actually ends up being neater.

Reply:
You can at least leave three arguments blank, without creating the extra goal.  That's definitely better (and same number of lines).  Please do at least that -- I agree that reasonable people can differ on whether the four-argument-blank version is better.